common/web/recorder/build.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+#
+# Compiles development-related KeymanWeb resources for use with developing/running tests.
+#   - the Recorder module (for engine tests)
+#   - the DOM module (for touch-alias and element-interface tests)
+
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+################################ Main script ################################
+
+
+builder_describe \
+  "Compiles the web-oriented utility function module." \
+  configure \
+  clean \
+  build \
+  ":module      Builds recorder-core module" \
+  ":proctor     Builds headless-testing, node-oriented 'proctor' component"
+
+builder_parse "$@"
+
+# START - Script parameter configuration
+REPORT_STYLE="local"  # Default setting.
+
+if builder_has_option --ci; then
+  REPORT_STYLE="ci"
+
+  echo "Replacing user-friendly test reports with CI-friendly versions."
+fi
+
+# END - Script parameter configuration
+
+CONFIGURE=
+if builder_has_action configure :module; then
+  CONFIGURE=true
+fi
+
+if builder_has_action configure :proctor; then
+  CONFIGURE=true
+fi
+
+if [[ $CONFIGURE == "true" ]]; then
+  verify_npm_setup
+
+  "$KEYMAN_ROOT/common/web/keyman-version/build.sh"
+
+  builder_report success configure :module
+  builder_report success configure :proctor
+fi
```

Comment:
```suggestion

do_configure() {
  verify_npm_setup
  "$KEYMAN_ROOT/common/web/keyman-version/build.sh"
}

CONFIGURE=false
if builder_has_action configure :module; then
  CONFIGURE=true
  do_configure
  builder_report success configure :module
fi

if builder_has_action configure :proctor; then
  if !$CONFIGURE; then
    do_configure
  else
    echo All configuration completed in configure:proctor, not repeating.
  fi
  builder_report success configure :proctor
fi
```

An alternative way to skin this cat (not tested) ... which would avoid the nesting of build actions, which I think is preferable, because they are not really nested actions, just identical ones.

I can see this being further improved by having `builder_has_action` returning 1/0 with no side-effects, and `builder_start_action` actually starting the action, for example:

```
if builder_has_action configure :module || builder_has_action configure :proctor; then
  builder_start_action configure :module 
  verify_npm_setup
  "$KEYMAN_ROOT/common/web/keyman-version/build.sh"
  builder_report success configure :module
fi
```

(But that's a bigger change so no need to do it here...)

Reply:
On that 'later idea', can you add `builder_has_action_quiet` which is the simple test, and I'll do a follow-up PR to rename it and `builder_has_action`? (and perhaps also `builder_report` to `builder_finish_action`?)

Reply:
See #7333; I figured I'd put that as the start of a new draft PR.  Feel free to commandeer the PR from there if you like.